pkg/openebs/mayaproxy/mayaproxy.go

```diff
@@ -0,0 +1,159 @@
+package mayaproxy
+
+import (
+	"encoding/json"
+	"github.com/golang/glog"
+	"net/http"
+	"bytes"
+	"io/ioutil"
+	mayav1 "github.com/princerachit/csi-openebs/pkg/openebs/v1"
+	"gopkg.in/yaml.v2"
+	"errors"
+	"net/url"
+)
+
+// CreateVolume requests mapi server to create an openebs volume. It returns an error if volume creation fails
+func (mayaService *MayaService) CreateVolume(mapiURI *url.URL, spec mayav1.VolumeSpec) error {
+	// Marshal serializes the value provided into a YAML document
+	yamlValue, _ := yaml.Marshal(spec)
+
+	glog.Infof("[DEBUG] volume Spec Created:\n%v\n", string(yamlValue))
+
+	url, err := mayaService.GetVolumeURL(mapiURI, versionLatest)
+	glog.Infof("[DEBUG] create volume URL %v", url.String())
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("POST", url.String(), bytes.NewBuffer(yamlValue))
+	req.Header.Add("Content-Type", "application/yaml")
+	c := &http.Client{
+		Timeout: timeout,
+	}
+	resp, err := c.Do(req)
+
+	if err != nil {
+		glog.Errorf("Error when connecting maya-apiserver %v", err)
+		return err
+	}
+	defer resp.Body.Close()
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		glog.Errorf("Unable to read response from maya-apiserver %v", err)
+		return err
+	}
+	code := resp.StatusCode
+	if code != http.StatusOK {
+		glog.Errorf("Error response from maya-apiserver: %v", http.StatusText(code))
+		return errors.New("error response from maya-apiserver")
+	}
+
+	responseBody := string(data)
+	glog.Infof("volume Successfully Created:\n%v\n", responseBody)
+	return nil
+}
+
+// DeleteVolume requests mapi server to delete an openebs volume. It returns an error if volume deletion fails
+func (mayaService *MayaService) DeleteVolume(mapiURI *url.URL, volumeName string) error {
+	glog.Infof("[DEBUG] Delete Volume :%v", string(volumeName))
```

Comment:
better to use log level here for debug logs, for example `glog.V(2).Infof("----")` .